util/dynamicConfig.js

```diff
@@ -1,24 +1,38 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
+function recurse(event, value, Config, field) {
+  value = value.replace("{{", "");
+  value = value.replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const getFieldVal = get(event, path);
+    if (getFieldVal) {
+      Config[field] = getFieldVal;
+      unset(event, path);
+    } else {
+      Config[field] = JSON.parse(value.split("||")[1].trim()).toString();
+    }
+  }
+}
+
 function getDynamicConfig(event) {
   const { Config } = event.destination;
   if (Config) {
     Object.keys(Config).forEach(field => {
-      let value = Config[field].toString().trim();
-      if (value.startsWith("{{") && value.endsWith("}}")) {
-        value = value.replace("{{", "");
-        value = value.replace("}}", "");
-        if (value.includes("||")) {
-          const path = value.split("||")[0].trim();
-          const getFieldVal = get(event, path);
-          if (getFieldVal) {
-            Config[field] = getFieldVal;
-            unset(event, path);
-          } else {
-            Config[field] = JSON.parse(value.split("||")[1].trim()).toString();
+      const value = Config[field].toString().trim();
+      if (Array.isArray(value)) {
```

Comment:
Is there a possibility that `Config[field]` can be an object in any case ?

Reply:
No I don't think we can have that case. @ruchiramoitra

Reply:
you are checking only if it is not string. this would not work if the value is a boolean. Check the other scenarios too.

Reply:
like normal string not matching the patter that value is passed. boolean should also go as it is

Reply:
Have updated the check to look for `object` instead of !== "string" . js treats array as objects only. Therefore.

Reply:
IMO the only data types we should avoid are `function` and `symbol`. All other combinations should be supported.

Reply:
Updated to support `objects` as well